src/test/java/jenkins/security/plugins/ldap/LDAPDNEscapingTest.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.security.plugins.ldap;
+
+import java.io.InputStream;
+import java.util.Collections;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.JenkinsRule.WebClient;
+import org.springframework.security.core.userdetails.UserDetails;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import hudson.security.GroupDetails;
+import hudson.security.LDAPSecurityRealm;
+import hudson.util.Secret;
+import jenkins.model.IdStrategy;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Tests {@link LDAPConfiguration} with DNs that are not URL safe.
+ */
+@LDAPTestConfiguration
+@RunWith(JUnitParamsRunner.class)
+public class LDAPDNEscapingTest {
+
+    @ClassRule
+    public static LDAPRule ads = new LDAPRule();
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+
+    @BeforeClass
+    public static void setupLdap() throws Exception {
+        try (InputStream ldifIs = LDAPDNEscapingTest.class.getResourceAsStream("/jenkins/security/plugins/ldap/LDAPDNEscapingTest/dnWithSpaces.ldif")) {
+            ads.loadSchema("planetexpress", "dc=planet express,dc=com", ldifIs);
+       }
+    }
+
+    // here as eclipse can not run a single parameterized test as it does not support the custom JUnitParamsRunner
+    //@Test
+    public void testSpacesInDN() throws Exception {
+        testOrgEscaping("dc=planet express,dc=com", null, null);
+    }
```

Comment:
```suggestion
    /*
    @Test
    public void testSpacesInDN() throws Exception {
        testOrgEscaping("dc=planet express,dc=com", null, null);
    }
    */
```

Reply:
does this matter?  at least in eclipse uncommenting the test would be twice as much work :-p

Reply:
Just for clarity, so that there is not an unused method lying around.